src/components/TimeRangePicker/TimeRangePicker.js

```diff
@@ -0,0 +1,24 @@
+import { useState } from "react";
+import { TimeRangePickerBody } from "./TimeRangePickerBody";
+import "./index.css";
+import { MiniClockIcon } from "../../icons/mini-clock";
+
+export const TimeRangePicker = () => {
+  let [isPickerOpen, setIsPickerOpen] = useState(false);
+
+  return (
+    <div className="relative">
+      <div
+        className="time-range-picker-widget flex items-center justify-center px-2 py-1 bg-gray-50 cursor-pointer rounded-sm border border-gray-300"
+        onClick={() => setIsPickerOpen(!isPickerOpen)}>
```

Comment:
If the new state value depends from the old state value then use a callback in the setState function 
```suggestion
        onClick={() => setIsPickerOpen((prevIsPickerValue) => !prevIsPickerValue)}>
```